2_process/src/process_and_style.R

```diff
@@ -1,10 +1,39 @@
 process_data <- function(nwis_data){
-  nwis_data_clean <- rename(nwis_data, water_temperature = X_00010_00000) %>% 
+  
+  data_out <- data.frame(agency_cd = c(), site_no = c(), dateTime = c(), 
+                         X_00010_00000 = c(), X_00010_00000_cd = c(), tz_cd = c())
+  # loop through files to download 
+  for (download_file in nwis_data$filepath){
+    these_data <- read_csv(download_file, col_types = 'ccTdcc')
+    data_out <- rbind(data_out, these_data)
+  }
+
+  nwis_data_clean <- rename(data_out, water_temperature = X_00010_00000) %>% 
     select(-agency_cd, -X_00010_00000_cd, -tz_cd)
   
-  return(nwis_data_clean)
+  site_no <- unique(data_out$site_no)
+  site_info <- dataRetrieval::readNWISsite(site_no)
+  
+  annotated_data <- left_join(nwis_data_clean, site_info, by = "site_no") %>% 
+    select(station_name = station_nm, site_no, dateTime, water_temperature, latitude = dec_lat_va, longitude = dec_long_va) %>% 
+    mutate(station_name = as.factor(station_name))
+  
+  return(annotated_data) 
+}
+
+
+combine_into_df = function(...){
+  
+  site_files = c(...) 
+  
+  out_df = tibble(filepath = site_files, 
+                  hash = scipiper::get_remake_status(site_files)$hash)
```

Comment:
Huh, I suppose this probably does work here! I think I went with `tools::md5sum()` here, myself, and in the next course we'll show you the use of `sc_indicate`/`combine_to_ind` so you don't have to write a `combine_into_df` function yourself anyway...but given that this function depends on scipiper-complete site_files already, I think your approach would be as robust as anything else.

Reply:
Ah yeah, that makes sense. And it would be the same hash if using `tools::md5sum()` vs. `get_remake_status()$hash`, right?